kernel/aster-nix/src/fs/utils/page_cache.rs

```diff
@@ -198,6 +198,17 @@ impl Pager for PageCacheManager {
 
         Ok(())
     }
+
+    fn commit_overwrite(&self, idx: usize) -> Result<Frame> {
+        if let Some(page) = self.pages.lock().get(&idx) {
+            return Ok(page.frame.clone());
+        }
+
+        let page = Page::alloc_zero()?;
+        let frame = page.frame().clone();
+        self.pages.lock().put(idx, page);
+        Ok(frame)
```

Comment:
See #921 for the related concurrency bug fix and correct this.

Reply:
Fixed.